engines/extpoints/interfaces.go

```diff
@@ -32,4 +32,4 @@ type EngineOptions struct {
 // Any error here will be fatal and likely cause the worker to stop working.
 // If an implementor can determine that the platform isn't supported at
 // compile-time it is recommended to not register the implementation.
-type EngineProvider func(options EngineOptions) (engines.Engine, error)
+type EngineProvider func(options *EngineOptions) (engines.Engine, error)
```

Comment:
@petemoore, I haven't coded much go, maybe I'm wrong to suggest this pattern... It just seems like a  good way to solve the problem of adding additional input paramters... Without introducing something that is inherently complicated.

Note: This should all be documented in comments inside `EngineOptions` and similar options structs.